### PR TITLE
#14588 Use C_invoice.DueDate all over sql functions and reports 

### PR DESCRIPTION
**Pull request by cristinamghita**

### CODE REVIEW

File: backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/openitems_report.sql [hunk 1]
```diff
@@ -4,138 +4,138 @@ DROP TABLE IF EXISTS de_metas_endcustomer_fresh_reports.OpenItems_Report;
...
+            i.AD_Client_ID,
+            i.DocumentNo,
+            i.C_BPartner_ID,
+            i.IsSOTrx,
+            i.DateInvoiced,
+            i.DateAcct,
+            EXISTS (SELECT 0 FROM C_PaySelectionLine psl
+                                      JOIN C_PaySelection ps ON psl.C_PaySelection_ID = ps.C_PaySelection_ID AND ps.isActive = 'Y'
+                    WHERE ps.docstatus IN ('CO','CL') AND psl.C_Invoice_ID = i.C_Invoice_ID AND psl.isActive = 'Y') AS IsInPaySelection,
+            COALESCE ( p.NetDays, DaysBetween( i.DueDate::timestamp with time zone, i.DateInvoiced::timestamp with time zone ) ) AS NetDays,
```
Comment (metas-ts) [inline backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/openitems_report.sql:98]: is this right? in `C_Dunning_Candidate_Invoice_v1.sql` you have a `coalesce(ips.DueDate, i.DueDate)`

Comment (cristinamghita) [reply backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/openitems_report.sql:98]: @metas-ts Yes, I was in doubt here, but  Pay Schedule still shall be able to be used.

Comment (metas-ts) [reply backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/openitems_report.sql:98]: >  Yes, I was in doubt here, but Pay Schedule still shall be able to be used.

yes, fully agree..so why do you not have a similar `coalesce(ips.DueDate, i.DueDate)` in `openitems_report.sql?

Comment (cristinamghita) [reply backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/openitems_report.sql:98]: yes, see https://github.com/metasfresh/metasfresh/blob/5ae0c25ac21230e8a383b7815dbcf5f6eefdada8/backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/openitems_report.sql#L103

**Review by metas-ts — APPROVED** [review]

**Pull request merged**